libvast/src/system/exporter.cpp

```diff
@@ -227,11 +231,23 @@ behavior exporter(stateful_actor<exporter_state>* self, expression expr,
                    "ID set(s) in", runtime);
         if (self->state.accountant)
           self->send(self->state.accountant, "exporter.hits.runtime", runtime);
-        shutdown(self);
       }
+      if (finished(self->state.query))
+        shutdown(self);
+    },
+    [=](table_slice_ptr slice) {
+      handle_batch(to_events(*slice, self->state.hits));
     },
-    [=](std::vector<event>& candidates) {
-      handle_batch(candidates);
+    [=](done_atom, const caf::error& err) {
+      auto sender = self->current_sender();
+      if (sender == self->state.archive) {
+        if (err)
+          VAST_DEBUG(self, "received error from archive:",
+              self->system().render(err));
+        ++self->state.query.lookups_complete;
+      }
+      if (finished(self->state.query))
```

Comment:
Shouldn't this move into the previous if-block?

Reply:
I prefer it outside of the archive specific part of the handler. When the index sends a done for a request that produced no hits, the exporter would also be finished, right?

Reply:
True. Currently, the INDEX sends a different message but we might consolidate this in the future.